### ISSUE TITLE
What is the purpose of isView() inside Vue on Rails?

### ISSUE THREAD

**Issue by navjeetc** [ytbryan/vuejs#9]
What is the purpose of the statement `isView("###") `in the vue component with turbolinks support? What do I need to make sure is present in my erb markup to work with this?

**Comment by ytbryan** [comment]
Do you have the code/repo online? Can I take a look?

**Comment by navjeetc** [comment]
I do not have a repo as this is a project is for federal government. I will try to post a snippet of code from office if I can.

**Comment by ytbryan** [comment]
TLDR: If you are not using Specific-page Vue(SPV), you can remove isView(). And your turbolinks-enabled pack will still work.

## The purpose
isView() is part of the Specific-Page Vue (SPV) approach to selectively show your JS packs on specific Rails view. It is designed with Rails 6's default JS compiler in mind. 

## Usage
The following block of code will only run on `pages#index` page. This assumes Rails 6's webpacker as its default javascript compiler. Replace `###` with Rails `controller#action` route naming convention.

``` javascript
if isView(“pages#index”){
   //only run on pages#index
    const something = new Vue({
      render: h => h(App)
    }).$mount('#something') //this will look for id="something" or <%= vue_component :something %>
    console.log({something})
}
```

## To enable Specific-Page Vue

If you are using Rails 5, you will need to enable specific-page-vue by doing the following: 

1. At `app/assets/javascripts/application.js`
``` javascript
//= require vue-on-rails
//= require_tree .
```

2. At your `app/views/layouts/application.html.erb` (Rails 6 should have *_pack_tag by default)
```
    <%= javascript_pack_tag 'application' %>
    <%= stylesheet_pack_tag 'application' %>
  </head>
```

```
  <body class="<%= specific_page_vue %>">
    <%= yield %>
  </body>
```
> Note: When you jumpstart your Vue on Rails with the application template `-m http://vueonrails.com/vue`, you only need to enable `vue-on-rails` at your `app/assets/javascripts/application.js` and the rest is done for you.

Hope this is useful for you

**Comment by navjeetc** [comment]
That is indeed very helpful. Thanks  a lot.
